code/game/turfs/simulated/wall_attacks.dm

```diff
@@ -69,27 +69,25 @@
 
 	try_touch(user, rotting)
 
-/turf/simulated/wall/attack_generic(var/mob/user, var/damage, var/attack_message, var/wallbreaker)
+
+
+/turf/simulated/wall/attack_generic(mob/M, damage, attack_message)
+	M.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
 
 	radiate()
-	if(!istype(user))
+	if(!istype(M))
 		return
 
-	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
-	var/rotting = (locate(/obj/effect/overlay/wallrot) in src)
-	if(!damage || !wallbreaker)
-		try_touch(user, rotting)
-		return
+	var/rot = locate(/obj/effect/overlay/wallrot) in src
+	var/hardness = reinf_material ? max(material.hardness, reinf_material.hardness) : material.hardness
 
-	if(rotting)
-		return success_smash(user)
+	if(!damage)
+		try_touch(M, rot)
 
-	if(reinf_material)
-		if((wallbreaker == 2) || (damage >= max(material.hardness,reinf_material.hardness)))
```

Comment:
I couldn't find many items or mobs that had the wallbreaker perk, there's already code for dealing with attacks from item and I don't think there's any cases where a thing would break walls quickly if it's not a high-damage/explosive thing and it's not a specialized thing like a q-del tool or a RCD